apps/db-service/src/index.ts

```diff
@@ -32,30 +35,82 @@ function getIdFromServerName(serverName: string) {
   return id
 }
 
+const PostgresErrorCodes = {
+  ConnectionException: '08000',
+} as const
+
+function sendFatalError(connection: PostgresConnection, code: string, message: string): never {
+  connection.sendError({
+    severity: 'FATAL',
+    code,
+    message,
+  })
+  connection.socket.end()
+  throw new Error(message)
+}
+
 async function fileExists(path: string): Promise<boolean> {
   try {
-    await access(path);
-    return true;
+    await access(path)
+    return true
   } catch {
-    return false;
+    return false
   }
 }
 
+const supabase = createClient(supabaseUrl, supabaseKey)
```

Comment:
Yes, I'm always on the type-safety side! :D 
The types are currently located in the `postgres-new` app, it might be worth making a dedicated "types" package so we don't break apps/packages boundaries when importing them.

Reply:
I've make a `@postgres-new/supabase` package so we can share the types easily now!